### ISSUE TITLE
Fonts: x48ng doesn't starts without arguments

### ISSUE THREAD

**Issue by MatMoul** [gwenhael-le-moine/x48ng#2]
Firstly, thanks to revive x48...
I sent a small PR for the x48ng.desktop file.

But on my setup I have a problem that was already there with x48.
When I run x48 or x48ng, I have to add the following arguments:
` -smallFont fixed -mediumFont fixed -largeFont fixed -connFont fixed`

When I launch it without args in console, I have this return :

```
x48ng: using XShm extension.
x48ng: FATAL ERROR, exit.
  - can't load font '-*-fixed-bold-r-normal-*-14-*-*-*-*-*-iso8859-1'
  - Please change resource 'smallLabelFont'

```

I suspect a missing font.

Do you have an idea?

**Comment by gwenhael-le-moine** [comment]
Thank you, it's a modest effort as I'm slowly getting into the codebase and I haven't written C for a long time now…

Indeed I think you're missing the font.

What does `xlsfonts | grep fixed | grep "iso8859-1$"` returns? (I guess nothing)

Does `xlsfonts | grep fixed | grep "iso10646-1$"` returns something?

I'm running Slackware which tends to be conservative and all-batteries-included but maybe your OS doesn't include such legacy fonts?

**Comment by gwenhael-le-moine** [comment]
You can check more directly with `xlsfonts -fn "-*-fixed-medium-r-normal-*-14-*-*-*-*-*-iso8859-1"`

**Comment by MatMoul** [comment]
Thanks for your feedback,
Result of the first cmd :
```
-misc-fixed-medium-r-semicondensed--0-0-75-75-c-0-iso8859-1
-misc-fixed-medium-r-semicondensed--13-100-100-100-c-60-iso8859-1
-misc-fixed-medium-r-semicondensed--13-120-75-75-c-60-iso8859-1
```
And the second cmd :
```
xlsfonts: pattern "-*-fixed-medium-r-normal-*-14-*-*-*-*-*-iso8859-1" unmatched
```
It seem that I'm missing the 14.
My XOrg installation is very minimal.
I didn't even have xlsfonts installed.

I will try to investigate these next days.

**Comment by MatMoul** [comment]
That's it, I found what I was missing on my ArchLinux installation.

Here is the list of dependencies I need:
readline libxext xterm xorg-mkfontscale xorg-fonts-misc

For info, I have published the x48ng and x48ng-git packages on [AUR](https://aur.archlinux.org/packages?O=0&SeB=nd&K=x48ng&outdated=&SB=p&SO=d&PP=50&submit=Go).
If you want to take control of it, ask me and I'll give it to you.

I maintain them using [arch-pkgbuilder](https://github.com/MatMoul/arch-pkgbuilder) and that's what allowed me to see your work so quickly.

**Comment by JohnRobson** [comment]
If you have this problem:
```
$ x48ng
FATAL ERROR, exit.
  - can't load font '-*-fixed-bold-r-normal-*-14-*-*-*-*-*-iso8859-1'
  - Please change resource 'x48ng'
```
try this:
```
x48ng --x11 --small-font "*-monotype-arial-bold-r-normal-*-12-*-*-*-*-*-iso10646-1" --medium-font "*-monotype-arial-medium-r-normal-*-14-*-*-*-*-*-iso10646-1" --large-font "*-monotype-arial-medium-r-normal-*-18-*-*-*-*-*-iso10646-1" --connection-font "*-monotype-arial-bold-r-normal-*-12-*-*-*-*-*-iso10646-1"
```

**Comment by hpsaturn** [comment]
Hi. First to all, **amazing project. Nice job, awesome!**. I loved so much programing in the Hp48 when I was young (in the ~1995 with the [nickname avp98](https://www.hpcalc.org/search.php?query=avp98)).

Well, I have [working it perfectly in my Debian 11](https://x.com/hpsaturn/status/1847423288114340170), but with Raspbian I have this issue. It is the same version, Bullseye Debian 11. I tried with the suggestions posted here. Thanks in advance.

**Comment by hpsaturn** [comment]
I notice that the ncurses version was working.. At the beginning I though that not, because the screen was clean, but I had need a missing "memory recovery" :)
But the x11 SDL version still without working..

**Comment by gwenhael-le-moine** [comment]
Hi, thank you for the kind words.

The SDL UI (`--sdl2`) doesn't need external fonts so it should work.

For the X11 UI (`--x11`) if you don't have the default "Fixed" you can try to change to fonts in `~/.config/x48ng/config.lua` to
```lua
font_small = "-*-noto mono-medium-r-normal-*-14-*-*-*-*-*-iso8859-1"
font_medium = "-*-noto mono-medium-r-normal-*-15-*-*-*-*-*-iso8859-1"
font_large = "-*-noto mono-medium-r-normal-*-20-*-*-*-*-*-iso8859-1"
font_devices = "-*-noto mono-medium-r-normal-*-12-*-*-*-*-*-iso8859-1"
```

**Comment by hpsaturn** [comment]
you are right, I don't know I believed that these options was producing the same error but not. I though that I tested it, but maybe I tested it with more parameters.. sorry. I confirm that with --sdl2 is working well (A little slowly rendering), and much better with x11 without any problem.  I notice that some shortcuts doesn't work with these options, for instance the ] and [, but maybe is something in my shortcuts.. Also with ncurses the ENTER and ESC keys doesn't work..
